sdk/src/metrics/aggregation/histogram_aggregation.cc

```diff
@@ -16,21 +18,22 @@ namespace metrics
 {
 
 LongHistogramAggregation::LongHistogramAggregation(
-    const HistogramAggregationConfig<long> *aggregation_config)
+    std::shared_ptr<AggregationConfig> aggregation_config)
 {
-  if (aggregation_config && aggregation_config->boundaries_.size())
+  auto ac = std::dynamic_pointer_cast<HistogramAggregationConfig>(aggregation_config);
```

Comment:
Can we use static_pointer_case here, or use dynamic cast only if RTTI macro is enabled. Something similar to what we do here - https://github.com/open-telemetry/opentelemetry-cpp/blob/c7dfc0de26128bedcdedcb5369ab33aa24aeb907/sdk/test/metrics/meter_provider_sdk_test.cc#L85

Reply:
I'm not sure yet if I want to keep the shard_ptr. The main issue was in the lambda that was passing the pointer.

Reply:
using static cast now.